### PR TITLE
[JENKINS-66854] `AsyncPeriodicWork` should not touch its log file unless it has something to say

### PR DESCRIPTION
**Pull request by jglick**
See [JENKINS-66854](https://issues.jenkins-ci.org/browse/JENKINS-66854).

Note that e.g. `BackgroundGlobalBuildDiscarder` will currently print something every hour if there are any jobs defined. But now `TelemetryReporter` will never create a log file.

Using `RewindableRotatingFileOutputStream` would make sense for frequently-run tasks, but I did not bother with that here.

### Proposed changelog entries

* Reduce the amount of disk writes to `logs/tasks/\*.log` when nothing interesting is happening.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
